nuv/util.go

```diff
@@ -127,3 +128,41 @@ func WriteFileToNuvolarisConfigDir(filename string, content []byte) (string, err
 	}
 	return path, nil
 }
+
+// kind does not work using the socat proxy that VSCode introduced so it needs a workaround when running nuv
+func patchDockerHost() error {
+	fsys := os.DirFS("/")
+	// if .dockerenv exists
+	info, err := fs.Stat(fsys, ".dockerenv")
+	if os.IsNotExist(err) {
+		return err
+	}
+	// and is a regular file
+	if !info.Mode().IsRegular() {
+		return fmt.Errorf(".dockerenv exists but it's not a regular file")
+	}
+
+	// and if docker-host.sock exists
+	info, err = fs.Stat(fsys, "var/run/docker-host.sock")
+	if os.IsNotExist(err) {
+		return err
+	}
+	// and is a socket
+	if info.Mode().Type() != fs.ModeSocket {
+		return fmt.Errorf("docker-host.sock exists but it's not a socket")
+	}
+
+	// and $DOCKER_HOST is empty
+	if len(os.Getenv("DOCKER_HOST")) > 0 {
+		return fmt.Errorf("$DOCKER_HOST not empty")
+	}
+
+	// then sudo env DOCKER_HOST=unix:///var/run/docker-host.sock nuv devcluster"
+	// os.Setenv("DOCKER_HOST", "unix:///var/run/docker-host.sock")
+	cmd := exec.Command("sudo", "env", "DOCKER_HOST=unix:///var/run/docker-host.sock")
```

Comment:
here you should just re-execute sudo env DOCKER_HOST and append os.Args() so it re-execute itself under sudo with the env variable changed